notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb

```diff
@@ -0,0 +1,631 @@
+    "        self._sum_score += cur_score\n",
+    "        self._sum_words += cur_words\n",
+    "        self._cur_score = cur_score / cur_words\n",
+    "\n",
+    "        result = cur_score / cur_words if cur_words != 0 else 0\n",
+    "        return result\n",
+    "    \n",
+    "    def _decode(self, logits):\n",
+    "        token_ids = np.squeeze(np.argmax(logits, -1))\n",
+    "        tokens = [self._decoding_vocab[idx] for idx in token_ids]\n",
+    "        tokens = [token_group[0] for token_group in groupby(tokens)]\n",
+    "        tokens = [t for t in tokens if t != self.pad_token]\n",
+    "        res_string = ''.join([t if t != self.words_delimiter else ' ' for t in tokens]).strip()\n",
+    "        res_string = ' '.join(res_string.split(' '))\n",
+    "        res_string = res_string.lower()\n",
+    "        return res_string\n",
+    "\n",
+    "    def _editdistance_eval(self, source, target):\n",
+    "        n, m = len(source), len(target)\n",
+    "\n",
+    "        distance = np.zeros((n+1, m+1), dtype=int)\n",
+    "        distance[:, 0] = np.arange(0, n+1)\n",
+    "        distance[0, :] = np.arange(0, m+1)\n",
+    "\n",
+    "        for i in range(1, n + 1):\n",
+    "            for j in range(1, m + 1):\n",
+    "                cost = 0 if source[i - 1] == target[j - 1] else 1\n",
+    "\n",
+    "                distance[i][j] = min(distance[i - 1][j] + 1,\n",
+    "                                    distance[i][j - 1] + 1,\n",
+    "                                    distance[i - 1][j - 1] + cost)\n",
+    "        return distance[n][m]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9bbbca4a",
+   "metadata": {
+    "id": "CclWk-fVd9Wi"
+   },
+   "source": [
+    "## Run Quantization Pipeline\n",
+    "Here we define a configuration for our quantization pipeline and run it. Please note that we use built-in `IEEngine` implementation of `Engine` interface from the POT API for model inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "16457bd4",
+   "metadata": {
+    "id": "PiAvrwo0tr6Z",
+    "tags": [],
+    "test_replace": {
+     "\"stat_subset_size\": 300,": "\"stat_subset_size\": 3,"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "model_config = Dict({\"model_name\": \"wav2vec2_base\", \"model\": ir_model_xml, \"weights\": ir_model_bin})\n",
+    "engine_config = Dict({\"device\": \"CPU\"})\n",
+    "dataset_config = {\n",
+    "    \"data_source\": os.path.join(DATA_DIR, \"LibriSpeech/test-clean\"),\n",
+    "}\n",
+    "algorithms = [\n",
+    "    {\n",
+    "        \"name\": \"DefaultQuantization\",\n",
+    "        \"params\": {\n",
+    "            \"target_device\": \"ANY\",\n",
+    "            \"model_type\": \"transformer\",\n",
+    "            \"preset\": \"performance\",\n",
+    "            \"stat_subset_size\": 300,\n",
+    "            \"activations\": {\n",
+    "                \"range_estimator\": {\n",
```

Comment:
How is this configuration was found for this model? Is it gives us a more accurate model in comparison with the default configuration?

Reply:
Particular layers are not currently processed automatically during quantization which end up in dramatical accuracy drop, so they are ignored for now as a work-around.

Reply:
'activation' section is required for now, significant accuracy reduction without it. '214' layer ignore is WA for known issue. That's it, the rest is default.